com.microsoft.mrtk.core/Utilities/Async/Internal/AsyncCoroutineRunner.cs

```diff
@@ -0,0 +1,175 @@
+// MIT License
+
+// Copyright(c) 2016 Modest Tree Media Inc
+
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+using System;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+using UnityEngine;
+
+[assembly: InternalsVisibleTo("Microsoft.MixedReality.Toolkit.Tests.PlayModeTests")]
+namespace Microsoft.MixedReality.Toolkit.Utilities
+{
+    /// <summary>
+    /// This Async Coroutine Runner is just an object to
+    /// ensure that coroutines run properly with async/await.
+    /// </summary>
+    /// <remarks>
+    /// <para>The object that this MonoBehavior is attached to must be a root object in the
+    /// scene, as it will be marked as DontDestroyOnLoad (so that when scenes are changed,
+    /// it will persist instead of being destroyed). The runner will force itself to
+    /// the root of the scene if it's rooted elsewhere.</para>
```

Comment:
This seems... kind of gross? Do we really need these systems? Especially in the Core package that basically everyone (even all of our internal partner teams) consume?

Reply:
Removed this package